server/src/main/java/org/opensearch/search/profile/ContextualProfileBreakdown.java

```diff
@@ -20,17 +20,18 @@
  *
  * @opensearch.internal
  */
-public abstract class ContextualProfileBreakdown<T extends Enum<T>> extends AbstractProfileBreakdown<T> {
-    public ContextualProfileBreakdown(Class<T> clazz) {
-        super(clazz);
+public abstract class ContextualProfileBreakdown extends AbstractProfileBreakdown {
+
+    public ContextualProfileBreakdown(Map<String, Class<? extends ProfileMetric>> metrics) {
+        super(metrics);
     }
 
     /**
      * Return (or create) contextual profile breakdown instance
      * @param context freestyle context
      * @return contextual profile breakdown instance
      */
-    public abstract AbstractProfileBreakdown<T> context(Object context);
+    public abstract ContextualProfileBreakdown context(Object context);
```

Comment:
I believe `AbstractProfileBreakdown` should be enough here, right?

```suggestion
    public abstract AbstractProfileBreakdown context(Object context);
```